vars/npmExecuteEndToEndTests.groovy

```diff
@@ -0,0 +1,128 @@
+        def script = checkScript(this, parameters) ?: this
+        def stageName = parameters.stage ?: env.STAGE_NAME
+
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixinStageConfig(script.commonPipelineEnvironment, stageName, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        // telemetry reporting
+        new Utils().pushToSWA([
+            step: STEP_NAME,
+            stepParamKey1: 'scriptMissing',
+            stepParam1: parameters?.script == null
+        ], config)
+
+        def e2ETests = [:]
+        def index = 1
+
+        def npmParameters = [:]
+        npmParameters.dockerOptions = ['--shm-size 512MB']
+
+        if (!config.appUrls) {
+            error "[${STEP_NAME}] The execution failed, since no appUrls are defined. Please provide appUrls as a list of maps.\n"
+
+        }
+        if (!(config.appUrls instanceof List)) {
+            error "[${STEP_NAME}] The execution failed, since appUrls is not a list. Please provide appUrls as a list of maps. For example:\n" +
+                "appUrls: \n" + "  - url: 'https://my-url.com'\n" + "    credentialId: myCreds"
+        }
+        if (!config.runScript) {
+            error "[${STEP_NAME}] No runScript was defined."
+        }
+
+        for (int i = 0; i < config.appUrls.size(); i++) {
+            List credentials = []
+            def appUrl = config.appUrls[i]
+
+            if (!(appUrl instanceof Map)) {
+                error "[${STEP_NAME}] The element ${appUrl} is not of type map. Please provide appUrls as a list of maps. For example:\n" +
+                    "appUrls: \n" + "  - url: 'https://my-url.com'\n" + "    credentialId: myCreds"
+            }
+            if (!appUrl.url) {
+                error "[${STEP_NAME}] No url property was defined for the following element in appUrls: ${appUrl}"
+            }
+            if (appUrl.credentialId) {
+                credentials.add(usernamePassword(credentialsId: appUrl.credentialId, passwordVariable: 'e2e_password', usernameVariable: 'e2e_username'))
+            }
+
+            Closure e2eTest = {
+                Utils utils = new Utils()
+                utils.unstashStageFiles(script, stageName)
+                try {
+                    withCredentials(credentials) {
+                        if (appUrl.parameters) {
+                            if (appUrl.parameters instanceof List) {
+                                npmExecuteScripts(script: script, parameters: npmParameters, install: false, virtualFrameBuffer: true, runScripts: [config.runScript], scriptOptions: ["--launchUrl=${appUrl.url}", appUrl.parameters])
```

Comment:
I would guess that `["--launchUrl=${appUrl.url}", appUrl.parameters]` resolved to something like `["--laun...", ["a","b"]]` which is not a flat map

Reply:
Yes you are right. It has to be a list actually and the tests were wrong as well. I addressed it with the latest commit.